### PR TITLE
MGMT-4315 Api changes and code generation for additional state for checking disk performance

### PR DESCRIPTION
**Pull request by ori-amizur**
Since the way of invocation and handling disk performance results changes, for backward compatibility new structures will be defined and will be handled both in the agent.  After transition, the old one will be removed from the agent and from the assisted-service

/cc @avishayt

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -3091,6 +3094,10 @@ definitions:
         x-go-custom-tag: gorm:"type:text"
         type: string
         description: The configured NTP sources on the host.
+      disks_speed:
+        x-go-custom-tag: gorm:"type:text"
+        type: string
+        description: The result of disk performance executed prior to beginning installation.
```
Comment (avishayt) [inline swagger.yaml:3100]: Is this some encoding of the speed per disk?  Is this the best way to store it?
@filanov thoughts?

Comment (avishayt) [reply swagger.yaml:3100]: I was hoping for something more like this, whether it's stored in a column or table:
disk_info {
  [
    id: <id>
    speed_ms: 
    role: 
  ],
 ...
}
  ...

Comment (ori-amizur) [reply swagger.yaml:3100]: Can try to do it as postgres json column.  Not sure I will be able to define the scheme here.

Comment (ori-amizur) [reply swagger.yaml:3100]: Since we are using same struct to access db and for json serializing (swagger), it will be hard to use json column

Comment (filanov) [reply swagger.yaml:3100]: I'm not sure how to solve it. 
Maybe we need to refactor inventory into multiple tables? network/disks etc.

Comment (ronniel1) [reply swagger.yaml:3100]: refactoring into multiple tables is not relevant right now

@ori-amizur Why not add these fields (that @avishayt  mentioned ), in the swagger scheme?

Comment (ronniel1) [reply swagger.yaml:3100]: @filanov refactoring the inventory DB into multiple tables is not really an option right now, is it?

Comment (filanov) [reply swagger.yaml:3100]: No it's not, but if we continue to delay it we will never do it.

Comment (ori-amizur) [reply swagger.yaml:3100]: I think it is not in the scope of this task.  We can do it in a different epic.

**Comment by filanov** [comment]
We will have disk info in two places? inventory and disk info ?

File: swagger.yaml [hunk 2]
```diff
@@ -3011,6 +3011,8 @@ definitions:
         type: integer
       disk_size_gb:
         type: integer
+      disk_speed_threshold_ms:
```
Comment (avishayt) [inline swagger.yaml:3014]:
```suggestion
      installation_disk_speed_threshold_ms:
```

File: swagger.yaml [hunk 3]
```diff
@@ -3206,6 +3209,10 @@ definitions:
         x-go-custom-tag: gorm:"type:text"
         type: string
         description: The configured NTP sources on the host.
+      disks_info:
+        x-go-custom-tag: gorm:"type:text"
+        type: string
+        description: Additional information in disks, including speed.
```
Comment (avishayt) [inline swagger.yaml:3215]:
```suggestion
        description: Additional information about disks, formatted as JSON.
```

File: swagger.yaml [hunk 4]
```diff
@@ -3305,6 +3312,7 @@ definitions:
       - api-vip-connectivity-check
       - ntp-synchronizer
       - fio-perf-check
+      - disk-speed-check
```
Comment (avishayt) [inline swagger.yaml:3315]:
```suggestion
      - installation-disk-speed-check
```

**Comment by avishayt** [comment]
> We will have disk info in two places? inventory and disk info ?

Unfortunately yes.  We discussed this on a call and it was decided that this would be easier and more robust than adding things to inventory that the agent isn't setting.  Doing that requires merging on every update of the settings or inventory (we are already doing this today).

**Review by avishayt — APPROVED** [review]

**Pull request merged**